protoc_plugin/lib/src/code_generator.dart

```diff
@@ -131,9 +134,19 @@ class CodeGenerator {
         response.file.addAll(gen.generateFiles(config));
       }
     }
-    response.supportedFeatures = Int64(
-      CodeGeneratorResponse_Feature.FEATURE_PROTO3_OPTIONAL.value,
-    );
+    response.supportedFeatures =
+        Int64(CodeGeneratorResponse_Feature.FEATURE_PROTO3_OPTIONAL.value) |
+        Int64(CodeGeneratorResponse_Feature.FEATURE_SUPPORTS_EDITIONS.value);
+    response.minimumEdition = Edition.EDITION_PROTO2.value;
+    response.maximumEdition = Edition.EDITION_2024.value;
+
+    // The edition defaults should always stay synchronized with the
+    // supported edition range we report to protoc.  It's not clear that
+    // the BUILD file definitions are load-bearing though, so we
+    // explicitly set them above and assert that the two are equal.
+    assert(response.minimumEdition == editionDefaults.minimumEdition.value);
```

Comment:
I wonder if we should add a CI check for this as well? That `protoc.dart` / pluginFeatureSetDefaults is the same as what we're reporting here?

If you think this is useful feel free to track in an issue.

Reply:
Good call, I moved these checks to a test.

The next assertion (the line below this one) is actually incorrect, I had a typo there and compared max. edition with min. expected edition, but I was still able to build the package from scratch (with all the test protos) and pass all of the tests. So it turns out this code path is just never run with assertions enabled.

Which reminds me of this issue I opened years ago: #608. We should probably get back to it at some point.